tsdb/series_partition.go

```diff
@@ -255,13 +258,16 @@ func (p *SeriesPartition) CreateSeriesListIfNotExists(keys [][]byte, keyPartitio
 	}
 
 	// Check if we've crossed the compaction threshold.
-	if p.compactionsEnabled() && !p.compacting && p.CompactThreshold != 0 && p.index.InMemCount() >= uint64(p.CompactThreshold) {
+	if p.compactionsEnabled() && !p.compacting &&
+		p.CompactThreshold != 0 && p.index.InMemCount() >= uint64(p.CompactThreshold) &&
+		p.compactionLimiter.TryTake() {
```

Comment:
This looks to me like we will block a write if there is no room for a compaction.

Reply:
Notice the `TryTake()` method doesn't block:
```golang
// TryTake attempts to take a token and return true if successful, otherwise returns false.
func (t Fixed) TryTake() bool {
```